extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheMongoEntityEnhancer.java

```diff
@@ -70,6 +70,43 @@ protected void generateAccessorGetField(MethodVisitor mv, EntityField field) {
             mv.visitFieldInsn(Opcodes.GETFIELD, thisClass.getInternalName(), field.name, field.descriptor);
         }
 
+        @Override
+        public void visitEnd() {
```

Comment:
Why can';t you use the `@GenerateBridge` annotation for these two?

Reply:
It didn't generates the method signature that I needs, with it I have: `java.lang.NoSuchMethodError: 'com.mongodb.client.MongoDatabase io.quarkus.mongodb.panache.runtime.MongoOperations.mongoDatabase(java.lang.Class)'`

So I need to generate them by hand.

Reply:
I don't see why it does not. Can you use `javap` to figure out the difference in signature?

Reply:
Just found why, it's because the name didn't match : `mongoCollection()` and `getMongoCollection()`.

I remember implementing it with `@GenerateBridge` at first without success, I don't know why it works now but it works ;)

If you wonder why it's not `getMongoCollection()` the name of the method to retrieve the collection it's because it will be called automatically by JSON-B when serializing and Entity ...

Reply:
Well, you can add a `@BsonIgnore` annotation to it if you want. I'm not entirely sure if I prefer `getFoo()` or `foo()` getters, but it has to be consistent within an extension.

Reply:
`@BsonIgnore` is only used by Mongo not JSON-B nor Jackson so it wil not solve the issue here. And getting the uderlying collection/database is not like getting an entity field so I don't think it's important to be consistent with getters that are designed for entity fields.

Reply:
See what we do for `PanacheEntityBase.isTransient()` ;) We add ignore annotations at build time.

Reply:
It's for `PanacheEntityBase.isPersisten()`:) , I saw that you directly put `@JsonbTransient` and you have an optional dependency to JSON-B.
I'm OK with it but this should be an issue with Jackson ...

Reply:
Yes, sorry, `isPersistent()`. And you're right, my memory is really bad, we do not add an annotation at build time. For Jackson it's not an issue because it won't serialise it if it has no setter (at least that's what my commit message at the time said).

Reply:
Unfortunatly it didn't works because of the `AutomaticPojoCodec` that generates an error for this field even if I ignore it with `@BsonIgnore`: `Top level classes with generic types are not supported by the PojoCodec` for this to works we will need specialized codec for `MongoCollection` and `MongoDatabase` this is a lot of useless works (as these codecs will never be used).

So I propose to keep `mongoCollection()` and `mongoDatabase()` ... I can change the name to `retrieveMongoCollection/MongoDatabase` if you prefere.

Reply:
Hah, that's fine. If it can't be done because of BSON, keep the current names.